blockdao/blockdao.md

```diff
@@ -0,0 +1,63 @@
+# Block DAO
+Block DAO manages data access to the blockchain, for example reading a raw data block, querying a tx by hash, and committing a block to the blockchain
+
+The blockchain data consist of multiple block files (chain.db, chain-00000001.db, etc), the state trie db file (trie.db), and the index db (index.db)
+
+We'll be focusing on block file in the following
+
+## Legacy file (before db split activation at Greenland height)
+Each db file is a key-value db storing raw data in multiple buckets, using *block hash as the key*
+1. block header in bucket named "bhr"
+2. block body in bucket named "bbd"
+3. block footer in bucket named "bfr"
+4. block receipt in bucket named "rpt"
+
+In particular, the very first db file (`chain.db`) stores the additional information:
+1. the height and hash of the tip block is stored in bucket named "blk"
+2. the height/hash mapping of all blocks in bucket named "h2h"
+
+When committing a block, 2 things happen:
+1. block data are written into corresponding db file (like chain-00000001.db)
+2. height/hash of the new block, and height/hash mapping are written to `chain.db`
+
+## New file (after db split activation at Greenland height)
+The quest of a new file comes from 2 considerations:
+
+First, storing the height/hash mapping in the first file (chain.db) caused dependency of the block DAO service on this special file, we want to remove that dependency such that **the blockchain is able to start up and running from any block db file**
```

Comment:
does it mean - this node has chain-0000003.db that stores blocks from height 10K and it can start from 10K by ignoring 1 to 10K-1 blocks?

Reply:
that's right